src/main/java/frc/robot/commands/ReleaseCoral.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands;
+
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import com.spikes2212.dashboard.RootNamespace;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.subsystems.Storage;
+
+import java.util.function.Supplier;
+
+public class ReleaseCoral extends MoveGenericSubsystem {
+
+    private static final RootNamespace NAMESPACE = new RootNamespace("release coral");
+    private static final Supplier<Double> TIME_TO_RELEASE = NAMESPACE.addConstantDouble("time to release", 0.5);
+    private static final Supplier<Double> RELEASE_SPEED = NAMESPACE.addConstantDouble("release speed", 0.5);
+
+    private double startTime;
+
+    public ReleaseCoral(Storage storage) {
+        super(storage, RELEASE_SPEED);
+    }
+
+    @Override
+    public void initialize() {
+        startTime = Timer.getFPGATimestamp();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return Timer.getFPGATimestamp() - startTime >= TIME_TO_RELEASE.get();
```

Comment:
the timer should only start once the coral is out of the system

Reply:
And why is that?
Once the coral is out if the system the command should stop. And you dont really have a way to check if the coral is out, because your sensor is on the entrance. Plain old timer for thr whole command can do the trick 😉